packages/core/src/util/stringify.ts

```diff
@@ -50,3 +50,23 @@ export function concatStringsWithSpace(before: string|null, after: string|null):
       (after === null ? '' : after) :
       ((after == null || after === '') ? before : before + ' ' + after);
 }
+
+/**
+ * Ellipses the string in the middle when longer than the max length
+ *
+ * @param string
+ * @param maxLength of the output string
+ * @returns elispsed string with ... in the middle
+ */
+export function truncateMiddle(str: string, maxLength = 100): string {
+  if (!str) return str;
+  if (maxLength < 1) return str;
+  if (str.length <= maxLength) return str;
+  if (maxLength == 1) return str.substring(0, 1) + '...';
+
+  const midpoint = Math.ceil(str.length / 2);
+  const toremove = str.length - maxLength + 3;
+  const lstrip = Math.ceil(toremove / 2);
+  const rstrip = toremove - lstrip;
+  return str.substring(0, midpoint - lstrip) + '...' + str.substring(midpoint + rstrip);
```

Comment:
nit: we can probably simplify this function to:
```suggestion
  const halfLimit = Math.round(maxLength / 2);
  return str.substring(0, halfLimit) + '...' + str.substring(str.length - halfLimit);
```